http_job_queue.go

```diff
@@ -84,29 +84,31 @@ func (q *HTTPJobQueue) Jobs(ctx gocontext.Context) (outChan <-chan Job, err erro
 				time.Sleep(time.Second)
 			} else {
 				for _, id := range jobIds {
```

Comment:
This `jobIds` slice should only contain as many job IDs as there are ready processors, although in the case of a multi-source queue in front of an HTTP and AMQP queue, the ready processors count may be incorrect, which is why each ID is additionally guarded by the select over `<-ready`.